File: app/models/ai_resource.py
import uuid
from datetime import datetime
from typing import Optional, Dict, Any
from sqlmodel import SQLModel, Field
from sqlalchemy import Column, String
from sqlalchemy.dialects.sqlite import JSON as SQLiteJSON

class AiResource(SQLModel, table=True):
    __tablename__ = "ai_resources"

    id: uuid.UUID = Field(default_factory=uuid.uuid4, primary_key=True)
    name: str = Field(index=True)
    type: str = Field(index=True)  # text_llm, vision_llm, ocr_paddle, etc.
    endpoint: str
    api_key: Optional[str] = None  # In a real app, encrypt this
    config: Dict[str, Any] = Field(default={}, sa_column=Column(SQLiteJSON))
    is_enabled: bool = Field(default=True)
    is_default: bool = Field(default=False)
    description: Optional[str] = None
    health_status: str = Field(default="unknown")  # unknown, healthy, unhealthy
    last_health_check_at: Optional[datetime] = None
    created_at: datetime = Field(default_factory=datetime.utcnow)
    updated_at: datetime = Field(default_factory=datetime.utcnow)

File: frontend/src/types/aiResource.ts
export enum AiResourceType {
  TEXT_LLM = 'text_llm',
  VISION_LLM = 'vision_llm',
  OCR_PADDLE = 'ocr_paddle',
  OCR_DEEPSEEK = 'ocr_deepseek',
  EMBEDDING = 'embedding',
  RERANKER = 'reranker',
}

export const AiResourceTypeLabels: Record<string, string> = {
  [AiResourceType.TEXT_LLM]: 'Text LLM',
  [AiResourceType.VISION_LLM]: 'Vision LLM',
  [AiResourceType.OCR_PADDLE]: 'PaddleOCR',
  [AiResourceType.OCR_DEEPSEEK]: 'DeepSeek-OCR',
  [AiResourceType.EMBEDDING]: 'Embedding Model',
  [AiResourceType.RERANKER]: 'Reranker',
};

export interface AiResource {
  id: string;
  name: string;
  type: string;
  endpoint: string;
  api_key?: string; // Optional for display (masked)
  config: Record<string, any>;
  is_enabled: boolean;
  is_default: boolean;
  description?: string;
  health_status: 'unknown' | 'healthy' | 'unhealthy';
  last_health_check_at?: string;
  created_at: string;
  updated_at: string;
}

export interface AiResourceCreate {
  name: string;
  type: string;
  endpoint: string;
  api_key?: string;
  config?: Record<string, any>;
  is_enabled?: boolean;
  is_default?: boolean;
  description?: string;
}

export interface AiResourceUpdate {
  name?: string;
  type?: string;
  endpoint?: string;
  api_key?: string;
  config?: Record<string, any>;
  is_enabled?: boolean;
  is_default?: boolean;
  description?: string;
}
